File: NET/LatencyTesting/LatencyTesting_CSharp/EffectConfiguration.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Collections;

namespace LatencyTesting
{
    public class EffectConfiguration
    {
        private Hashtable properties;
        private int sampleRate, bufferSize;

        public EffectConfiguration(int _sampleRate, int _bufferSize, Hashtable _properties)
        {
            properties = _properties;
            sampleRate = _sampleRate;
            bufferSize = _bufferSize;
        }

        public String GetValueByKey(String _key)
        {
            String ret = null;
            IDictionaryEnumerator enumerator = properties.GetEnumerator();
            enumerator.Reset();
            while (enumerator.MoveNext())
            {
                if (enumerator.Entry.Key.ToString() == _key)
                {
                    ret = enumerator.Entry.Value.ToString();
                    break;
                }
            }             
            if(ret == null)
            {
                throw new EffectConfigurationException("No configuration properties exist with that name/key");
            }
            return ret;
        }

        public int BufferSize
        {
            get
            {
                return bufferSize;
            }
        }

        public int SampleRate
        {
            get
            {
                return sampleRate;
            }
        }

        public class EffectConfigurationException : Exception
        {
            public EffectConfigurationException(String msg)
                : base(msg)
            { }
        }
    }
}

File: NET/LatencyTesting/LatencyTesting_C++/EffectConfiguration.cpp
#include "stdafx.h"
#include "EffectConfiguration.h"

namespace Processor
{
	EffectConfiguration::EffectConfiguration(int sampleRate, int bufferSize, System::Collections::Hashtable ^properties)
	{
		_sampleRate = sampleRate;
		_bufferSize = bufferSize;
		_properties = properties;
	}

	String^ EffectConfiguration::GetValueByKey(System::String ^key)
	{
		String^ ret = nullptr;
		IDictionaryEnumerator^ enumerator = _properties->GetEnumerator();
		enumerator->Reset();
		while(enumerator->MoveNext())
		{
			DictionaryEntry ^entry = enumerator->Entry;
			if(entry->Key->ToString() == key->ToString())
			{
				ret = entry->Value->ToString();
				break;
			}
		}
		if(ret == nullptr)
		{
			throw gcnew EffectConfigurationException("Key does not exist in this Config object");
		}
		return ret;
	}

	int EffectConfiguration::BufferSize::get()
	{
		return _bufferSize;
	}

	int EffectConfiguration::SampleRate::get()
	{
		return _sampleRate;
	}

	EffectConfiguration::EffectConfigurationException::EffectConfigurationException(System::String ^message) : Exception(message)
	{}
}